core/src/main/java/com/graphhopper/GraphHopperReader.java

```diff
@@ -0,0 +1,138 @@
+package com.graphhopper;
+
+import com.graphhopper.routing.AlgorithmOptions;
+import com.graphhopper.routing.RoutingAlgorithmFactory;
+import com.graphhopper.routing.RoutingAlgorithmFactorySimple;
+import com.graphhopper.routing.ch.CHRoutingAlgorithmFactory;
+import com.graphhopper.routing.profiles.DefaultEncodedValueFactory;
+import com.graphhopper.routing.querygraph.QueryGraph;
+import com.graphhopper.routing.template.ViaRoutingTemplate;
+import com.graphhopper.routing.util.*;
+import com.graphhopper.routing.weighting.*;
+import com.graphhopper.storage.CHProfile;
+import com.graphhopper.storage.GHDirectory;
+import com.graphhopper.storage.Graph;
+import com.graphhopper.storage.GraphHopperStorage;
+import com.graphhopper.storage.index.LocationIndex;
+import com.graphhopper.storage.index.LocationIndexTree;
+import com.graphhopper.storage.index.QueryResult;
+import com.graphhopper.util.PathMerger;
+import com.graphhopper.util.TranslationMap;
+
+import java.io.File;
+import java.util.List;
+
+import static com.graphhopper.util.Helper.isEmpty;
+import static com.graphhopper.util.Helper.toLowerCase;
+
+public final class GraphHopperReader {
+
+    private final TranslationMap trMap = new TranslationMap().doImport();
+    private final EncodingManager encodingManager;
+    private final GraphHopperStorage graphHopperStorage;
+    private final CHPreparationHandler chHandler;
+    private final LocationIndex locationIndex;
+
+    public GraphHopperReader(GraphHopperWriter ghWriter) {
```

Comment:
A reader takes a writer :) hmm ;), but this seems to be a convenience constructor only?